AxonDeepSeg/data_management/dataset_building.py

```diff
@@ -78,43 +82,42 @@ def patched_to_dataset(path_patched_data, path_dataset, type_, random_seed=None)
     np.random.seed(random_seed)
 
     # First we define where we are going to store the patched data
-    if not os.path.exists(path_dataset):
-        os.makedirs(path_dataset)  
+    if not path_dataset.exists():
+        path_dataset.mkdir(parents=True)
 
     # First case: there is only one type of acquisition to use.
     if type_ == 'unique':
 
         i = 0 # Total patches index
 
         # We loop through all folders containing patches
-        for patches_folder in tqdm(os.listdir(path_patched_data)):
+        patches_folder_names = [f for f in path_patched_data.iterdir()]
+        for patches_folder in tqdm(patches_folder_names):
 
-            path_patches_folder = os.path.join(path_patched_data, patches_folder)
-            if os.path.isdir(path_patches_folder):
+            path_patches_folder = path_patched_data / patches_folder.name
+            if path_patches_folder.is_dir():
 
                 # We are now in the patches folder
                 L_img, L_mask = [], []
-                for data in os.listdir(path_patches_folder):
-
-                    data_name = data[:-4].split('_')
+                filenames = [f for f in path_patches_folder.iterdir()]
+                for data in filenames:
+                    root, index = data.stem.split('_')
 
-                    if 'image' in data:
+                    if 'image' in data.name:
+                        img = imread(path_patches_folder / data.name, flatten=True, mode='L')
+                        L_img.append((img, int(index)))
 
-                        img = imread(os.path.join(path_patches_folder, data), flatten=True, mode='L')
-                        L_img.append((img, int(data_name[-1])))
-
-                    elif 'mask' in data:
-
-                        mask = imread(os.path.join(path_patches_folder, data), flatten=True, mode='L')
-                        L_mask.append((mask, int(data_name[-1])))
+                    elif 'mask' in data.name:
+                        mask = imread(path_patches_folder / data, flatten=True, mode='L')
+                        L_mask.append((mask, int(index)))
 
                 # Now we sort the patches to be sure we get them in the right order
-                L_img_sorted, L_mask_sorted = sort_list_files(L_img, L_mask)
+                L_img, L_mask = sort_list_files(L_img, L_mask)
 
                 # Saving the images in the new folder
-                for img,k in L_img_sorted:
-                    imsave(os.path.join(path_dataset,'image_%s.png'%i), img, 'png')
-                    imsave(os.path.join(path_dataset,'mask_%s.png'%i), L_mask_sorted[k][0], 'png')
+                for img,k in L_img:
+                    imsave(path_dataset.joinpath('image_%s.png'%i), img, 'png')
```

Comment:
Same comment as above, maybe we should keep the standardize way of joining paths to be `path 1 \ path2` rather than using `Path`'s `joinpath` function; my example above could be used as a template.